apps/jubmoji-quest/src/pages/cardholder/index.tsx

```diff
@@ -0,0 +1,486 @@
+// @ts-ignore
+import { execHaloCmdWeb } from "@arx-research/libhalo/api/web.js";
+import { useEffect, useRef, useState } from "react";
+import MobileDetect from "mobile-detect";
+import { InfoModal } from "@/components/modals/InfoModal";
+import { AppHeader } from "@/components/AppHeader";
+import { Icons } from "@/components/Icons";
+import { PowerCard } from "@/components/cards/PowerCard";
+import { Message } from "@/components/Message";
+import { Modal, ModalProps } from "@/components/modals/Modal";
+import { Card } from "@/components/cards/Card";
+import { Button } from "@/components/ui/Button";
+import Image from "next/image";
+import { toast } from "react-hot-toast";
+import {
+  getRandomNullifierRandomness,
+  getMessageHash,
+  PublicMessageSignature,
+  createProofInstance,
+  cardPubKeys,
+  NfcCardRawSignature,
+} from "jubmoji-api";
+import { bigIntToHex } from "babyjubjub-ecdsa";
+import Link from "next/link";
+import { Textarea } from "@/components/ui/Textarea";
+
+enum CardholderActions {
+  EDIT_CARD_DETAILS,
+  PROPOSE_QUEST,
+  PSEUD_X_POST,
+  PSEUD_TELEGRAM_POST,
+  JOIN_TG_GROUP,
+}
+
+const cardholderActionDetails: Record<
+  CardholderActions,
+  {
+    title: string;
+    buttonText: string;
+    entryFields?: { field: string; size: "big" | "small" }[];
+    redirect?: string;
+  }
+> = {
+  [CardholderActions.EDIT_CARD_DETAILS]: {
+    title: "Cardholder edit portal",
+    buttonText: "Unlock editing",
```

Comment:
I don't think these should say "Unlock" - the phrasing is quite confusing. Think its better to just use directives such as "Begin Editing"

Reply:
good point fixed!